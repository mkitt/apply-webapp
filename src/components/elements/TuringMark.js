// @flow
import React from 'react'
import { always } from 'ramda'
import { BASE_URL } from '../../constants/networking'

const markSrc = always(`${BASE_URL}/turing-school-mark-256.png`)

type Props = {
  width?: number,
  height?: number,
}

const TuringMark = ({ width, height }: Props) => (
  <img
    alt="turing school mark"
    src={markSrc()}
    width={width}
    height={height}
  />
)

TuringMark.defaultProps = {
  width: 48,
  height: 48,
}

export default TuringMark
