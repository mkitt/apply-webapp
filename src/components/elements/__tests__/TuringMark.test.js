import React from 'react'
import renderer from 'react-test-renderer'
import TuringMark from '../TuringMark'

describe('<TuringMark />', () => {
  it('shows the correct snapshot tree for the <TuringMark> element', () => {
    const component = <TuringMark />
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the <TuringMark> element with options', () => {
    const component = (
      <TuringMark width={64} height={64} />
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
