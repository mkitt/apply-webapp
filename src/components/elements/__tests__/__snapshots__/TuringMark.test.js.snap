// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TuringMark /> shows the correct snapshot tree for the <TuringMark> element 1`] = `
<img
  alt="turing school mark"
  height={48}
  src="/turing-school-mark-256.png"
  width={48}
/>
`;

exports[`<TuringMark /> shows the correct snapshot tree for the <TuringMark> element with options 1`] = `
<img
  alt="turing school mark"
  height={64}
  src="/turing-school-mark-256.png"
  width={64}
/>
`;
